e2e/scenario_gpu_managed_experience_test.go

```diff
@@ -29,6 +30,81 @@ func getDCGMPackageNames(os string) []string {
 	return packages
 }
 
+// extractMajorMinorPatchVersion extracts the major.minor.patch version from a
+// version string
+//
+// Examples:
+//
+//	"4.6.0-1" -> "4.6.0"
+//	"4.5.2-1.azl3" -> "4.5.2"
+//	"1:4.4.1-1" -> "4.4.1" (handles epoch prefix)
+func extractMajorMinorPatchVersion(version string) string {
+	// Remove epoch prefix (e.g., "1:" in "1:4.4.1-1")
+	version = regexp.MustCompile(`^\d+:`).ReplaceAllString(version, "")
+
+	// Match major.minor.patch pattern
+	re := regexp.MustCompile(`^(\d+\.\d+\.\d+)`)
+	matches := re.FindStringSubmatch(version)
+	if len(matches) > 1 {
+		return matches[1]
+	}
+	return ""
+}
+
+type packageOSVariant struct {
+	pkgName   string
+	osName    string
+	osRelease string
+}
+
+func Test_Version_Consistency_GPU_Managed_Components(t *testing.T) {
```

Comment:
Should we keep this in a separate file (and maybe folder) since this is not an E2E test per se? More like component validation